config-service-api/src/testFixtures/java/org/hypertrace/config/service/test/MockGenericConfigService.java

```diff
@@ -114,6 +114,11 @@ public MockGenericConfigService withClock(Clock clock) {
     return this;
   }
 
+  public MockGenericConfigService withRequestContext(RequestContext context) {
```

Comment:
So this was really a placeholder context to avoid having to set a context in a test client, but if you actually do care about a context, it would be better to send it the regular way (via client) rather than assigning it in the server. Could we explore that option instead (that might look like changing the test interceptor in this file to only set the context if unset, for example).